src/modules/Registration/hooks/useCreateUser.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { toSignUpCredentials } from '@/api-types/signUpCredentials';
+import { AUTH_PAGE } from '@/constants/routes';
+import { useSignUpUserMutation } from '@/queries/auth';
+import { SignUpCredentialsForm } from '../types';
+
+type UseCreateUserParams = {
+  onError: () => void;
+};
+
+export const useCreateUser = ({ onError }: UseCreateUserParams) => {
+  const { mutate, isLoading } = useSignUpUserMutation();
+  const history = useHistory();
+
+  return {
+    createUser: useCallback(
+      (credentials: SignUpCredentialsForm) => {
+        // TODO: it should be moved to service's level
```

Comment:
I think it should namely in this place. Services must not know about View-model.